packages/common/src/types/filter.ts

```diff
@@ -387,4 +388,39 @@ export const compressDashboardFiltersToParam = (
         { dimensions: [], metrics: [], tableCalculations: [] },
     );
 
+export const isFilterRuleDefinedForFieldId = (
```

Comment:
was thinking we could make the steps a bit more explicit. What do you think?

```ts

export const isFilterRuleDefinedForFieldId = (
    filterGroup: FilterGroup,
    fieldId: string,
    isInterval: boolean = false,
) => {
    // Determine if the filter group is an 'and' or 'or' group
    const filterGroupItem = isAndFilterGroup(filterGroup)
        ? filterGroup.and
        : filterGroup.or;

    const isMatchingFieldId = (item: FilterGroupItem) => {
        if (!isFilterGroup(item)) {
            // If the item is not a filter group, check if it matches the fieldId
            return isInterval
                ? item.target.fieldId.startsWith(fieldId)
                : item.target.fieldId === fieldId;
        }
        return false;
    };

    const isFilterRulePresent = (
        item: OrFilterGroup | AndFilterGroup | FilterRule,
    ): boolean => {
        if (isMatchingFieldId(item)) {
            return true;
        }
        if (isFilterGroup(item)) {
            // If the item is a filter group, recursively check its items
            return isFilterRuleDefinedForFieldId(item, fieldId, isInterval);
        }
        return false;
    };

    return filterGroupItem.some(isFilterRulePresent);
};
```

Reply:
updated as suggested, thanks